libs/StatusGoQt/src/StatusGo/Chat/ChatDto.cpp

```diff
@@ -0,0 +1,241 @@
+            j.at(membersKey).get_to(d.members);
+        }
+        else if (j[membersKey].is_object()) {
+            auto obj = j[membersKey];
+            for (json::const_iterator it = obj.cbegin(); it != obj.cend(); ++it) {
+                ChatMember chatMember;
+                it.value().get_to(chatMember);
+                chatMember.id = it.key().c_str();
+                d.members.emplace_back(std::move(chatMember));
+            }
+        }
+    }
+}
+
+void Chats::to_json(json& j, const ChannelGroupDto& d) {
+    j = {{"id", d.id},
+         {"admin", d.admin},
+         {"verified", d.verified},
+         {"name", d.name},
+         {"description", d.description},
+         {"introMessage", d.introMessage},
+         {"outroMessage", d.outroMessage},
+         {"canManageUsers", d.canManageUsers},
+         {"color", d.color},
+         {"muted", d.muted},
+         {"images", d.images},
+         {"permissions", d.permissions},
+         {"channelGroupType", d.channelGroupType},
+         {"chats", d.chats},
+         {"categories", d.categories},
+         {"members", d.members},
+        };
+}
+
+void Chats::from_json(const json& j, ChannelGroupDto& d) {
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(admin)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(verified)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(name)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(description)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(introMessage)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(outroMessage)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(canManageUsers)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(color)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(muted)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(images)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(permissions, "permissions", false)
+
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(channelGroupType)
+    if (d.channelGroupType.isEmpty())
+        d.channelGroupType = ChannelGroupTypeUnknown;
+
+    constexpr auto chats = "chats";
+    if (j.contains(chats)) {
+        auto obj = j[chats];
+        for (json::const_iterator it = obj.cbegin(); it != obj.cend(); ++it) {
+            ChatDto chatDto;
+            it.value().get_to(chatDto);
+            d.chats.emplace_back(std::move(chatDto));
+        }
+    }
+
+    constexpr auto categories = "categories";
+    if (j.contains(categories)) {
+        auto obj = j[categories];
+        for (json::const_iterator it = obj.cbegin(); it != obj.cend(); ++it) {
+            Category category;
+            it.value().get_to(category);
+            d.categories.emplace_back(std::move(category));
+        }
+    }
+
+    constexpr auto membersKey = "members";
+    if (j.contains(membersKey)) {
+        if (j[membersKey].is_object()) {
+            auto obj = j[membersKey];
+            for (json::const_iterator it = obj.cbegin(); it != obj.cend(); ++it) {
+                ChatMember chatMember;
+                it.value().get_to(chatMember);
+                chatMember.id = it.key().c_str();
```

Comment:
Why not parsing automatically to a `map<ID, ChatMember>`? What is the advantage of vectorising it?

Reply:
@stefandunca it's not about advantage, but about the need and the thing we must do that way. Responses are not consistent, also nlohman doesn't iterate the json object, but only json array, but here we're receiving array of data as json object, check responses.

Reply:
It does parse the object if the assignment type is a map. It will map JSON object keys to map keys and corresponding values to map values. However, it looks here that we instead convert it to a vector and the key add it to the value of the entries as member data. And wondering what is the advantage?

Reply:
> And wondering what is the advantage?

@stefandunca check how the responses look in case of `chat_getChats` and `createCommunityChat` or `importCommunity` and you will see the difference. It's not about advantage, but about mapping the key from json object in one case and mapping the property of object from the array to the `id` of the `ChatMember`. That comes from an inconsistent responses from the status-go.